packages/mcp/src/tools/client/commit.ts

```diff
@@ -38,6 +39,55 @@ function populateDiffSpec(
 	args.push('--diff-spec', diffSpecJson);
 }
 
+/**
+ * Generates a summary string describing the reasons for rejected changes and the associated file paths.
+ *
+ * Groups the rejected changes by their reason and lists the file paths for each reason.
+ * The resulting string has the format: "reason1: file1, file2; reason2: file3".
+ *
+ * @param outcome - The outcome object containing paths to rejected changes, where each entry is a tuple of [reason, filePath].
+ * @returns A summary string listing reasons and their corresponding file paths.
+ */
+function createRejectionSummary(outcome: CreateCommitOutcome): string {
+	const groupedRejections = new Map<string, Set<string>>();
+	for (const path of outcome.pathsToRejectedChanges) {
+		const [reason, filePath] = path;
+		const paths = groupedRejections.get(reason) || new Set<string>();
+		paths.add(filePath);
+		groupedRejections.set(reason, paths);
+	}
+
+	const rejectionMessages = Array.from(groupedRejections.entries())
+		.map(([reason, paths]) => `${reason}: ${Array.from(paths).join(', ')}`)
```

Comment:
Since it's a `Set`, is the order in `paths` defined? `Array.from` probably won't do the sorting, and if that is true one might want to do it here.
The same might be true for the reasons, i.e. `groupedRejections.entries()`, which in practice would be less of an issue as it's unlikely there will be more than one, but possible.